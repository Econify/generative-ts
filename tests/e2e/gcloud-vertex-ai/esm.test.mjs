import * as esm from "../../../packages/gcloud-vertex-ai/dist/index.mjs";

describe('@generative-ts/gcloud-vertex-ai esm bundle', () => {
  test('api snapshot', async () => {
    expect(esm).toMatchSnapshot();
  });

  test('sanity check', async () => {
    const model = await esm.createVertexAiModelProvider({
      modelId: "gemini-1.0-pro",
      auth: {
        GCLOUD_LOCATION: 'us-central1',
        GCLOUD_PROJECT_ID: 'jmn-testing'
      }
    });

    const response = await model.sendRequest({
      prompt: "Say hi:",
      max_tokens: 50,
      temperature: 0
    });

    expect(response.data.candidates).toBeDefined();
  });
});
