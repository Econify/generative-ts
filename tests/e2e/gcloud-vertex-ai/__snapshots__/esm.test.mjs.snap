// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@generative-ts/gcloud-vertex-ai esm bundle api snapshot 1`] = `
{
  "createVertexAiModelProvider": [Function],
}
`;
